### PR TITLE
Add GovukNodes

### PR DESCRIPTION
**Pull request by elliotcm**
This provides functionality for fetching all DNS names for a given node class, regardless of whether on AWS or Carrenza.

Required for clearing varnish cache, since the `backend` machines need to issue calls directly to each `cache` box.

https://trello.com/c/f8U7EfUh/427-make-the-new-cache-clearing-app-invalidate-cache-in-varnish

### CODE REVIEW

File: lib/govuk_app_config/govuk_nodes/carrenza_fetcher.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    def hostnames_of_class(node_class)
+      instances_of_class(node_class).map do |instance|
+        instance.fetch("name")
+      end
+    end
+
+  private
+
+    def instances_of_class(node_class)
+      url = "http://puppetdb.cluster/v2/nodes?#{query_string(node_class)}"
```
Comment (deanwilson) [inline lib/govuk_app_config/govuk_nodes/carrenza_fetcher.rb]: Can this be provided by a env.d variable or some such? Hard coded names make me nervous.

Comment (elliotcm) [reply lib/govuk_app_config/govuk_nodes/carrenza_fetcher.rb]: It's hardcoded in the script I ported, but sure.  Do you have somewhere I can source it from in puppet, or should I hardcode it there?

Comment (elliotcm) [reply lib/govuk_app_config/govuk_nodes/carrenza_fetcher.rb]: Or I can put it in hiera but I'm assuming it's somewhere already

Comment (elliotcm) [reply lib/govuk_app_config/govuk_nodes/carrenza_fetcher.rb]: @deanwilson Updated.

**Comment by tijmenb** [comment]
I don't have any context on the project, but I can't figure out where this code will be used. Would it be possible to add some context to the README/docs about the purpose and intended usage?

**Comment by elliotcm** [comment]
@tijmenb sure

**Comment by elliotcm** [comment]
@tijmenb Updated.

**Comment by tijmenb** [comment]
@elliotcm thanks! If this will only be used by the cache clearing service, would it be better to put the code there initially? It would be good to keep govuk_app_config for app configuration, this seems more of a shared utility.

**Comment by elliotcm** [comment]
I had a feeling you were going to suggest that.  No problem, can do.

**Comment by elliotcm** [comment]
Incorporated into https://github.com/alphagov/cache-clearing-service/pull/6

**Pull request closed without merging**